src/renderers/shaders/ShaderChunk/clipping_planes_vertex.glsl

```diff
@@ -0,0 +1,4 @@
+#if NUM_USER_CLIPPING_PLANES > 0 && ! defined( STANDARD ) && ! defined( PHONG )
+	vViewPosition = - mvPosition.xyz;
```

Comment:
So we could have fragment shaders scanned and any variable they reference then gets automatically created in the vertex shader if it didn't exist previously?  Sort of like fulfilling dependences?  That may be a way to deal with some of this macro zoo.